internal/resources/akscluster/resource_akscluster.go

```diff
@@ -0,0 +1,235 @@
+func resourceClusterRead(ctx context.Context, data *schema.ResourceData, config any) diag.Diagnostics {
+	return dataSourceTMCAKSClusterRead(ctx, data, config)
+}
+
+// resourceClusterInPlaceUpdate updates AKS cluster settings in place then manually reconciles nodepool changes
+// updating, creating, and deleting where appropriate.
+func resourceClusterInPlaceUpdate(ctx context.Context, data *schema.ResourceData, config any) diag.Diagnostics {
+	tc, ok := config.(authctx.TanzuContext)
+	if !ok {
+		return diag.Errorf("error while retrieving Tanzu auth config")
+	}
+
+	clusterResp, nodepoolResp, getErr := getClusterAndNodepools(ctx, data, tc.TMCConnection)
+	if getErr != nil || clusterResp == nil || nodepoolResp == nil {
+		return diag.FromErr(errors.Errorf("Unable to get Tanzu Mission Control AKS cluster entry, name : %s", data.Get(NameKey)))
+	}
+
+	// Make changes to the cluster config.
+	if clusterChange := data.HasChange("spec.0.config.0"); clusterChange {
+		if updateErr := updateClusterConfig(ctx, data, clusterResp, tc); updateErr != nil {
+			return diag.FromErr(updateErr)
+		}
+	}
+
+	// Make changes to cluster nodepools.
+	if data.HasChange("spec.0.nodepool") {
+		if npChangeErr := handleNodepoolChanges(ctx, nodepoolResp.Nodepools, data, tc.TMCConnection); npChangeErr != nil {
+			return diag.FromErr(npChangeErr)
+		}
+	}
+
+	return dataSourceTMCAKSClusterRead(ctx, data, config)
+}
+
+// resourceClusterDelete deletes an AKS cluster and all associated node pools.
+func resourceClusterDelete(ctx context.Context, data *schema.ResourceData, config any) diag.Diagnostics {
+	tc, ok := config.(authctx.TanzuContext)
+	if !ok {
+		return diag.Errorf("error while retrieving Tanzu auth config")
+	}
+
+	fn := extractClusterFullName(data)
+	if err := tc.TMCConnection.AKSClusterResourceService.AksClusterResourceServiceDelete(fn, "false"); err != nil && !clienterrors.IsNotFoundError(err) {
+		return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control AKS cluster entry, name : %s", data.Get(NameKey)))
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, getTimeOut(data))
+	defer cancel()
+
+	if err := pollUntilClusterDeleted(ctx, data, tc.TMCConnection.AKSClusterResourceService, getPollInterval(ctx)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+// validate returns an error configuration will result in a cluster that will fail to create.
+func validate(nodepools []*models.VmwareTanzuManageV1alpha1AksclusterNodepoolNodepool) error {
+	for _, n := range nodepools {
+		if *n.Spec.Mode == models.VmwareTanzuManageV1alpha1AksclusterNodepoolModeSYSTEM {
+			return nil
+		}
+	}
+
+	return errors.New("AKS cluster must contain at least 1 SYSTEM nodepool")
+}
+
+// createOrUpdateCluster creates an AKS cluster in TMC.  It is possible the cluster already exists in which case the
+// existing cluster is updated with any node pools defined in the configuration.
```

Comment:
It would be possible for a cluster to have been created, and exist in a "PENDING" state in TMC, but was unable to transition to "READY" because the nodepool creation failed. This tries to account for that situation. using an existing partial resource rather than deleting and recreating it.